Attempt mpich 4.1.1 bump

milljm opened the pull request:
Add 'everything' package that installs everything. I use this to figure out future dependencies

Pin zfp in tools. Add vtk, zlib

~~Create new conda package for docs. Not sure how everyone is going to feel about this... Need to separate conda packages even more, with addition of other dependencies needed for latest PETSc (zfp <= 0.5.5) etc.~~
Do the opposite! Now that we know the offending package was by using `matplotlib` instead of `matplotlib-base`, we have determined we can instead remove `moose-test-tools`, and go back to a single 'tools' package.

~~Some documentation packages will attempt to bump this to 1.0.0. Some Peacock packages will also attempt to bump others to different values.~~ -not going to create a `moose-docs` package due to the previous findings of `matplotlib-base`.

Adjust Peacock dependencies so it works on WSL instances.

Download MUMPS manually, Skip tests. Seems to be an issue with downloading MUMPS, and then running the MUMPS test.

Closes #25598

Commit:
Attempt mpich 4.0.3 bump

Add 'everything' package that installs everything.
I use this to figure out future dependencies

Drop moose-test-tools

Now that we know matplotlib-base is ideally the package to be
installing, we can drop the need for having to separate our tool
packages.

Closes #25598

milljm commented:
This _will_ fail until I get a moose-docs Conda package recipe in place...

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1821551/) on c6da873 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1821551/) on c6da873 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1821551/) on c6da873 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1821757/) on e074c9c wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1821757/) on e074c9c wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1821757/) on e074c9c wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1821780/) on e074c9c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25743/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1821798/) on e074c9c wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>017740</th>
      <th colspan=4>#25743 e074c9</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.73%</td>
      <td>85.74%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>92096</td>
      <td>92100</td>
      <td>+4</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15328</td>
      <td>15324</td>
      <td>-4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25743/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25743/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25743/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1824594/) on 30262f1 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1824594/) on 30262f1 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1824594/) on 30262f1 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1824702/) on 1243a57 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1824702/) on 1243a57 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1824702/) on 1243a57 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Bumping versioner hashes

A change to a control package requires a bump to versioner_hashes.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1826994/) on a8d7f65 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1826994/) on a8d7f65 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1826994/) on a8d7f65 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Pin libclang in necessary packages

Linux        moose-mpich does not need libclang
Mac          moose-mpich needs libclang

Linux & Mac  moose-tools, moose-peacock needs libclang (clang-format, matplotlib, etc)

Because we don't want to introduce libclang in the Linux moose-mpich
package (Apptainer), we must therefor pin libclang in two separate
packages that use it.

WIP: I am betting another Linux moose- package needs libclang
(libmesh-vtk?, petsc?)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1827870/) on b46558d wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1827870/) on b46558d wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1827870/) on b46558d wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1828920/) on 40a6c22 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1828920/) on 40a6c22 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1828920/) on 40a6c22 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Adjust version strings

Still tweaking build strings to create a working dependency chain.

Fix 'everything' package to build for as many versions we support
Python.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829125/) on e4bc9bb wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829125/) on e4bc9bb wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829125/) on e4bc9bb wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour reviewed (changes requested):
please add cppunit-devel to `mpich.def` when you can

Commit:
Add cppunit-devel to mpich.def

This is needed for libMesh PRs

Commit:
Bump versioner_hashes

An underlying change requires a bump to our hash table.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829521/) on a0bb032 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829521/) on a0bb032 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829521/) on a0bb032 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Fix warnings.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829759/) on fe986ba wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829759/) on fe986ba wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1829759/) on fe986ba wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
Closing in favor of #26560
